cmake/CeleritasLibrary.cmake

```diff
@@ -7,869 +7,10 @@
-    set_target_properties(${target} PROPERTIES
-      # If cmake detects that a library depends/uses a static library
-      # linked with CUDA, it will turn CUDA_RESOLVE_DEVICE_SYMBOLS ON
-      # leading to a call to nvlink.  If we let this through (at least
-      # in case of Celeritas) we would need to add the DEVICE_LINK options
-      # also on non cuda libraries (that we detect depends on a cuda library).
-      # Note: we might be able to move this to celeritas_target_link_libraries
-      CUDA_RESOLVE_DEVICE_SYMBOLS OFF
-    )
-    if(NOT _target_type STREQUAL "EXECUTABLE")
-      get_target_property(_current_runtime ${target} CUDA_RUNTIME_LIBRARY)
-      if(NOT _current_runtime STREQUAL "Shared")
-        set(_need_to_use_shared_runtime FALSE)
-        foreach(_lib ${_alldependencies})
-          celeritas_check_cuda_runtime(_runtime ${_lib})
-          if(_runtime STREQUAL "Shared")
-            set(_need_to_use_shared_runtime TRUE)
-            break()
-          endif()
-        endforeach()
-        # We do not yet treat the case where the dependent library is Static
-        # and the current one is Shared.
-        if(${_need_to_use_shared_runtime})
-          set_target_properties(${target} PROPERTIES CUDA_RUNTIME_LIBRARY "Shared")
-        endif()
-      endif()
-    endif()
-  endif()
-
-endfunction()
-
-#-----------------------------------------------------------------------------#
-#
-# Return a flat list of all the direct and indirect dependencies of 'target'
-# which are libraries containing CUDA separatable code.
-#
-function(celeritas_cuda_gather_dependencies outlist target)
-  if(NOT TARGET ${target})
-    return()
-  endif()
-  get_target_property(_cached_dependencies ${target} CUDA_RDC_CACHED_LIB_DEPENDENCIES)
-  if (_cached_dependencies)
-     set(${outlist} ${_cached_dependencies} PARENT_SCOPE)
-     return()
-  endif()
-  celeritas_strip_alias(target ${target})
-  get_target_property(_target_type ${target} TYPE)
-  if(NOT _target_type STREQUAL "INTERFACE_LIBRARY")
-    get_target_property(_target_link_libraries ${target} LINK_LIBRARIES)
-    if(_target_link_libraries)
-      foreach(_lib ${_target_link_libraries})
-        celeritas_strip_alias(_lib ${_lib})
-        if(TARGET ${_lib})
-          celeritas_get_library_middle_target(_libmid ${_lib})
-        endif()
-        if(TARGET ${_libmid})
-          list(APPEND ${outlist} ${_libmid})
-        endif()
-        # and recurse
-        celeritas_cuda_gather_dependencies(_midlist ${_lib})
-        list(APPEND ${outlist} ${_midlist})
-      endforeach()
-    endif()
-  endif()
-  list(REMOVE_DUPLICATES ${outlist})
-  set_target_properties(${target} PROPERTIES CUDA_RDC_CACHED_LIB_DEPENDENCIES "${${outlist}}")
-  set(${outlist} ${${outlist}} PARENT_SCOPE)
-endfunction()
+include(CeleritasUtils.cmake)
 
-#-----------------------------------------------------------------------------#
+message(AUTHOR_WARNING "Deprecated file. Please use CeleritasUtils.cmake")
```

Comment:
Actually, don't we want people to only include `CudaRdcUtils` and use `cuda_rdc_add_library`? We *don't* want downstream code to add celeritas-specific aliases.

Reply:
Yes, that would be better.  In this case we don't need to distribute `CeleritasUtils.mk` but need to update the examples.